addons/block_code/locale/pt.po

```diff
@@ -0,0 +1,853 @@
+# res://addons/block_code/ui/blocks/entry_block/entry_block.gd
+# res://addons/block_code/ui/blocks/entry_block/entry_block.tscn
+# res://addons/block_code/ui/blocks/parameter_block/parameter_block.gd
+# res://addons/block_code/ui/blocks/parameter_block/parameter_block.tscn
+# res://addons/block_code/ui/blocks/statement_block/statement_block.gd
+# res://addons/block_code/ui/blocks/statement_block/statement_block.tscn
+# res://addons/block_code/ui/blocks/utilities/background/background.gd
+# res://addons/block_code/ui/blocks/utilities/background/gutter.gd
+# res://addons/block_code/ui/blocks/utilities/drag_drop_area/drag_drop_area.gd
+# res://addons/block_code/ui/blocks/utilities/drag_drop_area/drag_drop_area.tscn
+# res://addons/block_code/ui/blocks/utilities/parameter_input/parameter_input.gd
+# res://addons/block_code/ui/blocks/utilities/parameter_input/parameter_input.tscn
+# res://addons/block_code/ui/blocks/utilities/parameter_output/parameter_output.gd
+# res://addons/block_code/ui/blocks/utilities/parameter_output/parameter_output.tscn
+# res://addons/block_code/ui/blocks/utilities/snap_point/snap_point.gd
+# res://addons/block_code/ui/blocks/utilities/snap_point/snap_point.tscn
+# res://addons/block_code/ui/blocks/utilities/template_editor/template_editor.gd
+# res://addons/block_code/ui/blocks/utilities/template_editor/template_editor.tscn
+# res://addons/block_code/ui/picker/picker.gd
+# res://addons/block_code/ui/picker/picker.tscn
+# res://addons/block_code/ui/picker/categories/block_category.gd
+# res://addons/block_code/ui/picker/categories/block_category_button.gd
+# res://addons/block_code/ui/picker/categories/block_category_button.tscn
+# res://addons/block_code/ui/picker/categories/block_category_display.gd
+# res://addons/block_code/ui/picker/categories/block_category_display.tscn
+# res://addons/block_code/ui/picker/categories/category_factory.gd
+# res://addons/block_code/ui/picker/categories/variable_category/create_variable_button.gd
+# res://addons/block_code/ui/picker/categories/variable_category/create_variable_button.tscn
+# res://addons/block_code/ui/picker/categories/variable_category/create_variable_dialog.gd
+# res://addons/block_code/ui/picker/categories/variable_category/create_variable_dialog.tscn
+# res://addons/block_code/ui/picker/categories/variable_category/variable_category_display.gd
+# res://addons/block_code/ui/picker/categories/variable_category/variable_category_display.tscn
+# res://addons/block_code/ui/script_window/script_window.gd
+# res://addons/block_code/ui/script_window/script_window.tscn
+# res://addons/block_code/ui/title_bar/title_bar.gd
+# res://addons/block_code/ui/title_bar/title_bar.tscn
+# res://addons/block_code/ui/tooltip/tooltip.gd
+# res://addons/block_code/ui/tooltip/tooltip.tscn
+#
+# Automatically generated, 2024.
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: Block Coding Plugin\n"
+"Last-Translator: Automatically generated\n"
+"Language-Team: none\n"
+"Language: pt\n"
```

Comment:
You could ask a friendly Portuguese speaker to confirm but my understanding is that pt_BR and pt are different enough to warrant different translations.

Reply:
You're probably right. What I really want to avoid can be demonstrated with the `es_AR` locale. Since the editor has a generic `es` locale and the `es_AR` locale, it picks `es_AR` as the locale for any Spanish locale like `es_ES` or `es_MX`. It seems the only time `TranslationServer.get_tool_locale()` returns `es` is when I run with the bogus locale `LANG=es.utf-8`. So, I think that if we have the `es_AR` and `pt_BR`, then for all practical purposes the `es` and `pt` translations are unused. This seems like a bug in Godot - it should prefer the country-less locale if there's not a locale matching your country.

Reply:
That does seem like a bug!

If it picked `es_AR` for any latin-american Spanish locale but `es` for spanish-of-spain then that might make sense as a special-case because (IIUC) latin american dialects of spanish are more similar to one another than to spanish-of-spain...

Reply:
To demonstrate, save the following script:

```
@tool
extends EditorScript

func _run():
	var lang = OS.get_environment("LANG")
	var locale = TranslationServer.get_locale()
	var tool_locale = TranslationServer.get_tool_locale()
	print("LANG: %s" % lang)
	print("locale: %s" % locale)
	print("tool locale: %s" % tool_locale)
```

You can run this in the Script editor from its File menu or via the Ctrl+Shift+X shortcut. It's the closest thing to a REPL. Try that launching Godot with various `LANG` settings. You can also find out that the command line `--language` option only affects the game locale, not the tool locale.

Reply:
For example:
```
LANG: es_ES.utf-8
locale: es_AR
tool locale: es_AR
```

Reply:
On the other hand, since there's only a country-less `de` translation:
```
LANG: de_CH.utf-8
locale: de
tool locale: de
```
Or if there's no translation, it just returns the language:
```
LANG: en_GB.utf-8
locale: en
tool locale: en
```

Reply:
Ah, the bug appears to be introduced [here](https://github.com/godotengine/godot/commit/945645410905e8b23228965e29c2be274cb12645#diff-985565fb1f983acb2d87dbe8931c929d0f46fdecb1d99a31e495b151c0f52186R380) with the `>` change to `>=`. Both `es` and `es_AR` have a score of 1. `es` comes first in the list and previously would have remained as the best matched locale. The necessary part for https://github.com/godotengine/godot/issues/57125 is `score > 0` so that you don't use a locale that's totally unrelated.

Reply:
Looks like someone else hit this in https://github.com/godotengine/godot/issues/90677.

Reply:
I updated the commit message with information about this bug.